presidio-anonymizer/presidio_anonymizer/operators/operators_factory.py

```diff
@@ -1,19 +1,112 @@
 import logging
-from typing import Dict
+from typing import Dict, Type
 
 from presidio_anonymizer.entities import InvalidParamException
-from presidio_anonymizer.operators import OperatorType, Operator
+from presidio_anonymizer.operators import (
+    OperatorType,
+    Operator,
+    Custom,
+    Decrypt,
+    Encrypt,
+    Hash,
+    Keep,
+    Mask,
+    Redact,
+    Replace,
+    DeanonymizeKeep,
+)
+
+logger = logging.getLogger("presidio-anonymizer")
+
+# Predefined operators
+ANONYMIZERS = [Custom, Encrypt, Hash, Keep, Mask, Redact, Replace]
+DEANONYMIZERS = [Decrypt, DeanonymizeKeep]
 
 
 class OperatorsFactory:
     """Operators factory to get the correct operator class."""
 
-    _anonymizers: Dict = None
-    _deanonymizers: Dict = None
-    _operator_class: Dict = None
+    def __init__(
+        self,
+    ):
+        self._anonymizers = self.__load_predefined(OperatorType.Anonymize)
+        self._deanonymizers = self.__load_predefined(OperatorType.Deanonymize)
+
+    @staticmethod
+    def __load_predefined(operator_type: OperatorType) -> Dict[str, Type[Operator]]:
+        """
+        Return all predefined operators.
+
+        Return a dict with {name: operator class} for each operator.
+        """
+
+        operators = (
+            ANONYMIZERS if operator_type == OperatorType.Anonymize else DEANONYMIZERS
+        )
+        operators_dict = {
+            operator().operator_name(): operator for operator in operators
+        }
 
-    def __init__(self):
-        self.logger = logging.getLogger("presidio-anonymizer")
+        return operators_dict
+
+    @staticmethod
+    def __load_predefined_deanonymizers() -> Dict[str, Type[Operator]]:
+        """
+        Return all predefined deanonymizers.
+
+        Return a dict with name, deanonymizer class per each deanonymizer.
+        """
+
+        deanonymizers = [Decrypt, DeanonymizeKeep]
+
+        deanonymizers_dict = {
+            deanonymizer().operator_name(): deanonymizer
+            for deanonymizer in deanonymizers
+        }
+
+        return deanonymizers_dict
+
+    def add_anonymize_operator(self, operator: Type[Operator]):
+        """Add a new anonymizer to the factory.
+
+        :param operator: The operator class to add.
+        """
+        self._anonymizers[operator().operator_name()] = operator
+
+    def add_deanonymize_operator(self, operator: Type[Operator]):
+        """Add a new deanonymizer to the factory.
+
+        :param operator: The operator class to add.
+        """
+        self._deanonymizers[operator().operator_name()] = operator
+
+    def remove_anonymize_operator(self, operator: Type[Operator]):
```

Comment:
This could have been more modular (same logic for anonymizers and deanonymizers) but I figured it would be more straightforward to read this way, and I don't think we'll have a third operator type to support in the future. Happy to get feedback and update if needed.

Reply:
I don't think we need to go overboard with this and force two methods together just for the generalization of it. 
It's more clear to have them separately and as you said, there won't be a third operator :)